packages/tools/src/metamask.js

```diff
@@ -259,8 +259,37 @@ export class Metamask extends Emittery {
 
     const install = rpcPage.evaluate(
       async ({ snapId, version }) => {
-        const api = window.ethereum
+        // eslint-disable-next-line unicorn/consistent-function-scoping
+        const getRequestProvider = () => {
```

Comment:
@hugomrdias Had to add this in order to install correctly with rainbow. This is the same fix as https://github.com/filecoin-project/filsnap/pull/162. If we are shimming this in both places, how are we testing in `filecoin-project/filsnap` 🤔

Reply:
This is copied into this file instead of passing since you cannot pass non-serializable values https://playwright.dev/docs/evaluating#evaluation-argument

Reply:
whats the problem of shimming in both ?

Reply:
It's that the shimming in `filsnap` isn't actually being tested if we shim here also, so a regression in `filsnap` wouldn't actually be caught with `filsnap-testing`. I may be overthinking it, sometimes you just can't test every detail